server/src/main/java/org/cloudfoundry/identity/uaa/impl/config/YamlServletProfileInitializer.java

```diff
@@ -68,19 +63,21 @@
     public void initialize(ConfigurableWebApplicationContext applicationContext) {
 
         ServletContext servletContext = applicationContext.getServletContext();
+        final String contextPath = servletContext != null ? servletContext.getContextPath() : "/";
 
-        HttpSessionEventPublisher publisher = new HttpSessionEventPublisher();
-        servletContext.addListener(publisher);
+        if (servletContext != null) {
+            HttpSessionEventPublisher publisher = new HttpSessionEventPublisher();
+            servletContext.addListener(publisher);
```

Comment:
@joshuatcasey Am I correct in assuming that _pre-boot_ this is never null, and _post-boot_ it is always null? If so, I have a couple of questions/thoughts.

1) Once we switch to boot, how are we going to deal with the listener not getting added? It probably needs to be added elsewhere, right?
2) Since this condition is always satisfied pre-boot and never satisfied after boot, maybe we shouldn't put this in an `if` block.

Reply:
LGTM otherwise

Reply:
I believe that statement is correct with regards to when `servletContext` is/not `null`, so in Spring Boot that `if` statement will be dead code. 

In Spring Boot we can add a servlet listener like so:

https://www.concretepage.com/spring-boot/spring-boot-listener#ServletListenerRegistrationBean

Reply:
Also this might be relevant https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/web/servlet/ServletContextInitializer.html